astarte_device.c

```diff
@@ -145,6 +159,14 @@ astarte_device_handle_t astarte_device_init(astarte_device_config_t *cfg)
         xTaskNotify(ret->reinit_task_handle, NOTIFY_TERMINATE, eSetBits);
     }
 
+    if (ret->encoded_hwid) {
+        free(ret->encoded_hwid);
+    }
+
+    if (ret->realm) {
+        free(ret->realm);
+    }
```

Comment:
we don't need to check if ret->realm is null or not. free(NULL) is always no-op.